stdlib/Distributed/src/pmap.jl

```diff
@@ -153,7 +153,7 @@ function pmap(f, p::AbstractWorkerPool, c; distributed=true, batch_size=1, on_er
 end
 
 pmap(f, p::AbstractWorkerPool, c1, c...; kwargs...) = pmap(a->f(a...), p, zip(c1, c...); kwargs...)
-pmap(f, c; kwargs...) = pmap(f, default_worker_pool(), c; kwargs...)
+pmap(f, c; kwargs...) = pmap(f, CachingPool(), c; kwargs...)
```

Comment:
it might be worth explictly calling `clear!`
So as to know the cached functions would get marked for garbage collection.

```suggestion
function pmap(f, c; kwargs...)
    pool = CachingPool()
    try
        return pmap(f, pool, c; kwargs...)
    finally
        clear!(pool)
    end
end
```

Reply:
oops:
```suggestion
pmap(f, c; kwargs...) = pmap(f, CachingPool(workers()), c; kwargs...)
```